GUIDELINES.md

```diff
@@ -197,3 +197,94 @@ Instrumentation may add additional patch/unpatch messages for specific functions
 - In situations where the patch logic is not trivial and it helps to specify patch events in the right context and nuances. `aws-lambda` logs additional properties extracted from the lambda framework and exposes them for troubleshooting.
 
 The cases above are not covered by the base class and offer additional context to the user troubleshooting an issue with the instrumentation.
+
+## Supported Versions
+
+### Usage
+
+Instrumentation can specify the supported version to patch for each instrumented module and instrumented module file by specifying it in the `InstrumentationNodeModuleFile` and `InstrumentationNodeModuleDefinition` like so:
+
+```js
+const supporterVersions = ['>=1.2.3 <3'];
+
+  protected init() {
+
+    const someModuleFile = new InstrumentationNodeModuleFile(
+      'foo/lib/some-file.js',
+      supporterVersions,
+      myFilePatch,
+      myFileUnpatch,
+    );
+
+    const module = new InstrumentationNodeModuleDefinition(
+      'foo',
+      supporterVersions,
+      myModulePatch,
+      myModuleUnpatch,
+      [someModuleFile]
+    );
+    return module
+  }
+```
+
```

Comment:
The following feels like a lot of prose for the topic. I'm worried that if GUIDELINES get too long then they don't get read. Would you consider a (possibly much) shorter proposed alternative for this section?

Reply:
Sure I am open to making it shorter. 

Wondering if the guideline file is meant to be read fully by contributors, or more as a reference for a specific topic that documents our decisions and reasoning so it can be used for review, changes etc.

Feel free to suggest new shorter text 🙏

Reply:
I am torn on this - on one hand, all of this detail is really useful context. On the other, it indeed is a lot of text. What do you think about leading with the common example start to finish up front, then a lower section for "Variations" which can be linked from the common example for folks who know they have something different. A "tldr" does tend to be useful when there is a lot of content.

Reply:
I made changes to the GUIDELINES. making it shorter, moving stuff around. might not be perfect yet, but please have a second look when you get a chance